More architectures

I was trying to package lite-xl for Fedora Linux, but the build system fails when compiling for [ppc64le](https://kojipkgs.fedoraproject.org//work/tasks/1197/92941197/build.log) and [s390x](https://kojipkgs.fedoraproject.org//work/tasks/1198/92941198/build.log) architectures. Stating that
```
../src/main.c:107:6: error: #error "Please define -DLITE_ARCH_TUPLE."
  107 |     #error "Please define -DLITE_ARCH_TUPLE."
```

Sorry, but I'm not a developer, but I wonder if it would be possible to add

```
  #elif __s390x__
    #define ARCH_PROCESSOR "s390x"
  #elif __PPC64__
    #define ARCH_PROCESSOR "PPC64"
```
in addition to
https://github.com/lite-xl/lite-xl/blob/0fc793d1ae40ee1883aa6d8cac8659f48ee40e55/src/main.c#L85-L93

--- a/meson.build
+++ b/meson.build
@@ -52,6 +52,9 @@ lite_cargs = ['-DSDL_MAIN_HANDLED', '-DPCRE2_STATIC']
 if get_option('renderer') or host_machine.system() == 'darwin'
     lite_cargs += '-DLITE_USE_SDL_RENDERER'
 endif
+if get_option('arch_tuple') != ''
+    lite_cargs += '-DLITE_ARCH_TUPLE="@0@"'.format(get_option('arch_tuple'))
+endif
 #===============================================================================
 # Linker Settings
 #===============================================================================

--- a/meson_options.txt
+++ b/meson_options.txt
@@ -3,3 +3,4 @@ option('source-only', type : 'boolean', value : false, description: 'Configure s
 option('portable', type : 'boolean', value : false, description: 'Portable install')
 option('renderer', type : 'boolean', value : false, description: 'Use SDL renderer')
 option('dirmonitor_backend', type : 'combo', value : '', choices : ['', 'inotify', 'fsevents', 'kqueue', 'win32', 'dummy'], description: 'define what dirmonitor backend to use')
+option('arch_tuple', type : 'string', value : '', description: 'Specify a custom architecture tuple')

